aci/resource_aci_rest_managed.go

```diff
@@ -0,0 +1,310 @@
+
+		CreateContext: resourceAciRestManagedCreate,
+		UpdateContext: resourceAciRestManagedUpdate,
+		ReadContext:   resourceAciRestManagedRead,
+		DeleteContext: resourceAciRestManagedDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceAciRestManagedImport,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"id": {
+				Description: "The distinguished name of the object.",
+				Type:        schema.TypeString,
+				Computed:    true,
+			},
+			"dn": {
+				Type:        schema.TypeString,
+				Description: "Distinguished name of object being managed including its relative name, e.g. uni/tn-EXAMPLE_TENANT.",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"class_name": {
+				Type:        schema.TypeString,
+				Description: "Which class object is being created. (Make sure there is no colon in the classname)",
+				Required:    true,
+				ForceNew:    true,
+			},
+			"content": {
+				Type:        schema.TypeMap,
+				Description: "Map of key-value pairs those needed to be passed to the Model object as parameters. Make sure the key name matches the name with the object parameter in ACI.",
+				Optional:    true,
+				Computed:    true,
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					content := d.Get("content")
+					contentStrMap := toStrMap(content.(map[string]interface{}))
+					key := strings.Split(k, ".")[1]
+					if _, ok := contentStrMap[key]; ok {
+						return false
+					}
+					return true
+				},
+			},
+			"child": {
+				Type:        schema.TypeSet,
+				Description: "List of children.",
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"rn": {
+							Type:        schema.TypeString,
+							Description: "The relative name of the child object.",
+							Required:    true,
+						},
+						"class_name": {
+							Type:        schema.TypeString,
+							Description: "Class name of child object.",
+							Optional:    true,
+							Computed:    true,
+						},
+						"content": {
+							Type:        schema.TypeMap,
+							Description: "Map of key-value pairs which represents the attributes for the child object.",
+							Optional:    true,
+							Computed:    true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func getAciRestManaged(d *schema.ResourceData, c *container.Container) diag.Diagnostics {
+	className := d.Get("class_name").(string)
+	dn := d.Get("dn").(string)
+	d.SetId(dn)
+
+	content := d.Get("content")
+	contentStrMap := toStrMap(content.(map[string]interface{}))
+	newContent := make(map[string]interface{})
+	restContent, ok := c.Search("imdata", className, "attributes").Index(0).Data().(map[string]interface{})
+
+	if !ok {
+		return diag.Errorf("Failed to retrieve REST payload for class: %s.", className)
+	}
+
+	for attr, value := range restContent {
```

Comment:
what are these restContent and contentStrMap?

Reply:
`contentStrMap` is a map of attributes retrieved from TF state. `restContent` is a map of attributes as returned from the REST GET API call. Both of them are then used to compile an updated map of attributes (`newContent`).